### PR TITLE
[Ehn]Add rst version to origin command

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

When using `make_fit` and `dattorawacf` it adds the RST version into the `origin.command` at the end. 

**issue**: #422 

## Test

``` bash
make_fit 20180404.0200.12.lyr.rawacf > test.fitacf
dmapdump test.fitacf | grep origin.command | head -n 1
	string	"origin.command" = "make_fit 20180404.0200.12.lyr.rawacf RST Version: 4.5"
```

``` bash
dattorawacf 2005050520d.dat > test.rawacf
dmapdump test.rawacf | grep origin.command | head -n 1
	string	"origin.command" = "dattorawacf 2005050520d.dat RST Version: 4.5"
```

### CODE REVIEW

**Comment by pasha-ponomarenko** [comment]
@mts299, I have downloaded a zipped version of this branch, installed it and found that for some reason it is unstable with respect to `make_fit` and `dattorawacf` functionality. 

Sometimes these commands give this sort of error:
```
`ponomarenko@pasha-linux-main:~/work/origin_add_test> make_fit 20180201.0001.00.hok.rawacf > test_hok.fitacf
*** Error in `make_fit': malloc(): memory corruption (fast): 0x0000000000798a80 ***
Aborted (core dumped)`
```

On other ocassions they seem to work, however, the content of the string is corrupted (see extra stuff between the filename and RST version):

`ponomarenko@pasha-linux-main:~/work/origin_add_test> make_fit 20180201.0001.00.hok.rawacf > test_hok.fitacf
ponomarenko@pasha-linux-main:~/work/origin_add_test> dmapdump test_hok.fitacf | grep origin.command | head -n 1
        string  "origin.command" = "make_fit 20180201.0001.00.hok.rawacf▒Li RST Version: 4.5"`

I reinstalled this branch once more from scratch, but the problem remained.


For comparison, `develop` branch works properly, so it does not look like this is my local machine problem:
```
ponomarenko@pasha-linux-main:~/work/origin_add_test> make_fit 20180201.0001.00.hok.rawacf > test_hok.fitacf
ponomarenko@pasha-linux-main:~/work/origin_add_test> dmapdump test_hok.fitacf | grep origin.command | head -n 1
        string  "origin.command" = "make_fit 20180201.0001.00.hok.rawacf"
ponomarenko@pasha-linux-main:~/work/origin_add_test> dattorawacf 2000050102t.dat > test_t.rawacf
ponomarenko@pasha-linux-main:~/work/origin_add_test> dmapdump test_t.rawacf | grep origin.command | head -n 1
        string  "origin.command" = "dattorawacf 2000050102t.dat"
```

**Comment by mts299** [comment]
Thank you for testing @pasha-ponomarenko  I don't get the core dump so going to ask @ecbland to test it as well and then maybe need to do some pair programming to figure this out

**Comment by mts299** [comment]
fun I get it when I take `-g` option out which I need to debug

**Comment by pasha-ponomarenko** [comment]
@mts299, I suspect that something webnt wrong with zipping-unzipping the branch. How do you install yours, directly from github or from a zip file?

File: codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c [hunk 1]
```diff
@@ -81,23 +82,74 @@ int RadarParmSetOriginTime(struct RadarParm *ptr,char *str) {
...
+        free(ptr->origin.command);
     ptr->origin.command=NULL;
     return 0;
   }
 
-  if (ptr->origin.command==NULL) tmp=malloc(strlen(str)+1);
-  else tmp=realloc(ptr->origin.command,strlen(str)+1);
+  rst_version = GetRSTVersion();
+  if (ptr->origin.command==NULL) {
+      //fprintf(stderr, "before malloc\n");
```
Comment (ecbland) [inline codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c:141]:
```suggestion
```

File: codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c [hunk 2]
```diff
@@ -81,23 +82,74 @@ int RadarParmSetOriginTime(struct RadarParm *ptr,char *str) {
...
+      tmp=malloc(strlen(str)+strlen(rst_version));
+      //fprintf(stderr, "tmp extended\n");
```
Comment (ecbland) [inline codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c:143]:
```suggestion
```

File: codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c [hunk 3]
```diff
@@ -81,23 +82,74 @@ int RadarParmSetOriginTime(struct RadarParm *ptr,char *str) {
...
+  rst_path=getenv("RSTPATH");
+  strcpy(vname, rst_path);
+  strcat(vname, "/.rst.version");
+
+  vfp=fopen(vname, "r");
+
+  if (vfp != NULL) {
+    // read the .rst_version file 
+    while (fscanf(vfp,"%s",buff)==1); 
+    strcat(rst_version, " RST Version: ");
```
Comment (ecbland) [inline codebase/superdarn/src.lib/tk/radar.1.22/src/rprm.c:113]: Does it make sense to add some kind of delimiter between the command and the RST version number? No strong opinion here.
```suggestion
    strcat(rst_version, " -- RST Version: ");
```

**Review by ecbland — CHANGES_REQUESTED** [review]
I'm still getting core dump after that last commit. Using the same file as @pasha-ponomarenko 
```
make_fit 20180201.0001.00.hok.rawacf > test.fitacf
realloc(): invalid next size
Aborted (core dumped)

make_fit 20180201.0001.00.hok.rawacf > /dev/null
corrupted size vs. prev_size
Aborted (core dumped)
```
Notice the error is different depending on whether I write to an output fitacf file or to `/dev/null`

**Comment by mts299** [comment]
@ecbland k I will look into it again. Can we not suggest removing the prints until this is working :p I just committed them and then saw it was still core dumping so now I need them again. Normally I will double-check before saying merge haha

**Comment by mts299** [comment]
Going to close this PR this is a terrible idea in terms of memory issues. The problem is when origin.command is already set then we add a number of bytes but if we don't have the correct padding this corrupts things and it's a nightmare to deal with. 

I think the more elegant solution is adding an RST.version field as we have for radar and FITACF.

Plus I think that will make it more searchable for users then looking at the origin.command

**Comment by pasha-ponomarenko** [comment]
Looks like a good riddance!

**Pull request closed without merging**